### DISCUSSION TITLE
Google Crawls Invalid URLs from window.__remixContext

### DISCUSSION THREAD

**Discussion by xHomu** [remix-run/remix#5059] [Q&A] [answered]
On initial page load, Remix embeds loaderData inside script `window.__remixContext` for hydration:

![image](https://user-images.githubusercontent.com/84349818/211697184-c8ccd029-7fe4-425e-bb00-c2f57ce00f33.png)

It turns out, if you have any **relative** links in your loaderData, Google would index them as pathnames to be crawled later. That's how I end up with a thousand broken links according to Google Search Console:

![image](https://user-images.githubusercontent.com/84349818/211697382-b46e9525-b87c-41a7-bcd0-91f598334c57.png)

`/c/items/fortress-combat-engine-box-551` ends up as a broken link (The correct link is <https://nalu.wiki/neuralcloud/c/items/fortress-combat-engine-box-551>.

---

I'm not sure if this should be considered a Remix bug or not?

The obvious solution is to never send any relative links in loaderData in order to appease Google, but seems like an extra cognitive overhead for Remix developers. Is there any better solution? Am I the only one that encountered this?

**Comment by andreisena** [comment, marked as answer]
I don't think it is a Remix bug, but a known Google crawler behavior.
I would suggest you to add only /neuralcloud to allow list on your robots.txt file.
```
User-agent: *
Disallow: /
Allow: /neuralcloud
```

**Reply by xHomu** [reply]
My concern with this solution is it forces devs to update robot.txt every time a new top-level folder is added. 

Is there no way to flag robots to not crawl within <script> tags?

**Reply by andreisena** [reply]
Not that I know.
But you can generate robots.txt file dynamically creating a route `./app/routes/[robots.txt].tsx` and return the contents on the loader.
Or maybe check if the user-agent is a bot and do not render `<Scripts />` component in root.tsx file.

**Reply by xHomu** [reply]
Thanks for the suggestions! I'll play around with these ideas.

**Reply by sergiodxa** [reply]
> Or maybe check if the user-agent is a bot and do not render <Scripts /> component in root.tsx file.

I have been playing with this, using `isbot` package should be possible, but I'm not sure if Google can penalize you, doing this should also make the perf score better because you don't send any JS

**Reply by xHomu** [reply]
Looking forward to the article, Sergio!
